api/layer/layer.go

```diff
@@ -266,13 +268,20 @@ func (f MsgHandlerFunc) HandleMessage(ctx context.Context, msg *nats.Msg) error
 // NewLayer creates an instance of a layer. It checks credentials
 // and establishes gRPC connection with the node.
 func NewLayer(log *zap.Logger, neoFS NeoFS, config *Config) Client {
+	buffers := sync.Pool{}
+	buffers.New = func() any {
+		b := make([]byte, neoFS.MaxObjectSize())
```

Comment:
IMO, a potential OOM killer is here, the same way we already fixed this in SDK

Reply:
Using `sync.Pool` showed better memory using profile after load tests. It Already works here https://github.com/nspcc-dev/neofs-s3-gw/blob/4717bd84bdb8ed434012da7d0da0dff98947c46f/internal/neofs/neofs.go#L331

Reply:
> Links to changes, please

https://github.com/nspcc-dev/neofs-sdk-go/pull/539

> Using sync.Pool showed better memory using profile after load tests. It Already works here

yes, but putting 1K objects 1-byte sized simultaneously leads to 64GB memory usage but requires only 1KB in fact. we already know we may be quite successive at running out of _1TB memory_, not a cool story

but ok, as i understand we will return to this later